controllers/registry.go

```diff
@@ -192,64 +192,23 @@ func (r *DPAReconciler) ReconcileRegistries(log logr.Logger) (bool, error) {
 			},
 		}
-
-		if !dpa.BackupImages() {
-			deleteContext := context.Background()
-			if err := r.Get(deleteContext, types.NamespacedName{
-				Name:      registryDeployment.Name,
-				Namespace: r.NamespacedName.Namespace,
-			}, registryDeployment); err != nil {
-				if k8serror.IsNotFound(err) {
-					return true, nil
-				}
-				return false, err
-			}
-
-			deleteOptionPropagationForeground := metav1.DeletePropagationForeground
-			if err := r.Delete(deleteContext, registryDeployment, &client.DeleteOptions{PropagationPolicy: &deleteOptionPropagationForeground}); err != nil {
-				r.EventRecorder.Event(registryDeployment, corev1.EventTypeNormal, "DeleteRegistryDeploymentFailed", "Could not delete registry deployment:"+err.Error())
-				return false, err
-			}
-			r.EventRecorder.Event(registryDeployment, corev1.EventTypeNormal, "DeletedRegistryDeployment", "Registry Deployment deleted")
-
-			return true, nil
-		}
-
-		op, err := controllerutil.CreateOrUpdate(r.Context, r.Client, registryDeployment, func() error {
-
-			// Setting Registry Deployment selector if a new object is created as it is immutable
-			if registryDeployment.ObjectMeta.CreationTimestamp.IsZero() {
-				registryDeployment.Spec.Selector = &metav1.LabelSelector{
-					MatchLabels: map[string]string{
-						"component": registryName(&bsl),
-					},
-				}
-			}
-
-			err := controllerutil.SetControllerReference(&dpa, registryDeployment, r.Scheme)
-			if err != nil {
-				return err
+		deleteContext := context.Background()
+		if err := r.Get(deleteContext, types.NamespacedName{
+			Name:      registryDeployment.Name,
+			Namespace: r.NamespacedName.Namespace,
+		}, registryDeployment); err != nil {
+			if k8serror.IsNotFound(err) {
+				return true, nil
 			}
-			// update the Registry Deployment template
-			err = r.buildRegistryDeployment(registryDeployment, &bsl, &dpa)
-			return err
-		})
-
-		if err != nil {
 			return false, err
 		}
 
-		//TODO: Review registry deployment status and report errors and conditions
-
-		if op == controllerutil.OperationResultCreated || op == controllerutil.OperationResultUpdated {
-			// Trigger event to indicate registry deployment was created or updated
-			r.EventRecorder.Event(registryDeployment,
-				corev1.EventTypeNormal,
-				"RegistryDeploymentReconciled",
-				fmt.Sprintf("performed %s on registry deployment %s/%s", op, registryDeployment.Namespace, registryDeployment.Name),
-			)
+		deleteOptionPropagationForeground := metav1.DeletePropagationForeground
+		if err := r.Delete(deleteContext, registryDeployment, &client.DeleteOptions{PropagationPolicy: &deleteOptionPropagationForeground}); err != nil {
+			r.EventRecorder.Event(registryDeployment, corev1.EventTypeNormal, "DeleteRegistryDeploymentFailed", "Could not delete registry deployment:"+err.Error())
+			return false, err
 		}
-
+		r.EventRecorder.Event(registryDeployment, corev1.EventTypeNormal, "DeletedRegistryDeployment", "Registry Deployment deleted")
```

Comment:
Suggestion: Add the name and namespace of the registry deployment deleted in the Event instead of a generic statement.

Reply:
IIUC the object will be attached to the event anyways. I'll add name and namespace to the string.
```go
        // Event constructs an event from the given information and puts it in the queue for sending.
	// 'object' is the object this event is about. Event will make a reference-- or you may also
	// pass a reference to the object directly.
	// 'type' of this event, and can be one of Normal, Warning. New types could be added in future
	// 'reason' is the reason this event is generated. 'reason' should be short and unique; it
	// should be in UpperCamelCase format (starting with a capital letter). "reason" will be used
	// to automate handling of events, so imagine people writing switch statements to handle them.
	// You want to make that easy.
	// 'message' is intended to be human readable.
	//
	// The resulting event will be created in the same namespace as the reference object.
	Event(object runtime.Object, eventtype, reason, message string)
```